### DISCUSSION TITLE
Why need to add alpha_j and alpha_i in GATConv ?

### DISCUSSION THREAD

**Discussion by Nassiming** [pyg-team/pytorch_geometric#7161] [Q&A]
Hi, I'm a rookie to PyG. I'm confused about a line in the GATConv source code: 
```python
# we simply need to sum them up to "emulate" concatenation:
alpha = alpha_j if alpha_i is None else alpha_j + alpha_i
```

I don't understand why we need to add `alpha_j` and `alpha_i`. And I would greatly appreciate it if someone could provide me with additional explanations

**Comment by rusty1s** [comment]
`alpha_j` refers to the attention scores of the source node, and `alpha_i` refers to the attention scores of the destination nodes, which gets combined via summation. This is identical to first concatenation source and destination node features, and then applying multiplying that by an attention vector. However, this scheme is more efficient since we avoid the expensive concatenation here.
